a3p-integration/proposals/z:acceptance/vaults.test.js

```diff
@@ -0,0 +1,248 @@
+
+  await adjustVault(USER1ADDR, vaultID, { wantMinted: 1.0 });
+  await waitForBlock();
+
+  vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
+  const debtAfter = vaultData.debtSnapshot.debt.value;
+  t.log('vault debt after:', debtAfter);
+
+  t.is(
+    debtAfter,
+    debtBefore + 1005000n,
+    'The vault Debt should increase after removing some IST',
+  );
+});
+
+test.serial('close vault', async t => {
+  const activeVaults = await agopsVaults(USER1ADDR);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  let vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
+  const vaultCollateral = vaultData.locked.value;
+  t.log('vault collateral:', vaultCollateral);
+
+  const atomBalanceBefore = await getBalances([USER1ADDR], ATOM_DENOM);
+  t.log('atom balance before', atomBalanceBefore);
+
+  await closeVault(USER1ADDR, vaultID, 6.03);
+  await waitForBlock();
+
+  const atomBalanceAfter = await getBalances([USER1ADDR], ATOM_DENOM);
+  t.log('atom balance after', atomBalanceAfter);
+
+  vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
+  const vaultState = vaultData.vaultState;
+  t.log('vault state:', vaultState);
+
+  t.is(
+    atomBalanceAfter,
+    atomBalanceBefore + vaultCollateral,
+    'The ATOM balance should increase by the vault collateral amount',
+  );
+  t.is(vaultState, 'closed', 'The vault should be in the "closed" state.');
+});
+
+test.serial('open second vault', async t => {
+  const user2Address = await getUser('user2');
+  await bankSend(user2Address, `20000000${ATOM_DENOM}`);
+
+  const activeVaultsBefore = await agopsVaults(user2Address);
+  t.log('active vaults before:', activeVaultsBefore);
+
+  const mint = '7.0';
+  const collateral = '11.0';
+  await openVault(user2Address, mint, collateral);
+  await waitForBlock();
+
+  const activeVaultsAfter = await agopsVaults(user2Address);
+  t.log('active vaults after:', activeVaultsAfter);
+
+  t.is(
+    activeVaultsAfter.length,
+    activeVaultsBefore.length + 1,
+    `The number of active vaults should increase after opening a new vault.`,
+  );
+});
+
+test.serial('add collateral', async t => {
+  const user2Address = await getUser('user2');
+  const activeVaults = await agopsVaults(user2Address);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  let vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
+  const collateralBefore = vaultData.locked.value;
+  t.log('vault collateral before:', collateralBefore);
+
+  await adjustVault(user2Address, vaultID, { giveCollateral: 1.0 });
+  await waitForBlock();
+
+  vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
+  const collateralAfter = vaultData.locked.value;
+  t.log('vault collateral after:', collateralAfter);
+
+  t.is(
+    collateralBefore,
+    collateralAfter - 1000000n,
```

Comment:
using underscores to separate groups of digits is optional, but I find counting strings of zeros by eye to be error prone.

Javascript ignores underscores in numbers, so we often mimic the American English practice of separating triples with commas: 1,000,000.

```suggestion
    collateralAfter - 1_000_000n,
```

Reply:
Changes addressed on commit [4216ee1](https://github.com/Agoric/agoric-sdk/pull/10123/commits/4216ee1d9c560d7581f600009a8a9de321aa43d0)